### ISSUE TITLE
Swap the argument order of `apply_formatter_if`

### ISSUE THREAD

**Issue by carver** [pipermerriam/web3.py#382]
### What was wrong?

The function would read a little more naturally if the boolean test came first, instead of second.

### How can it be fixed?

* Switch the order in the function definition.
* Find all the usages and swap the arguments.

**Comment by wschwab** [comment]
I've done pretty much everything for this, but I have one or two questions:

In web3/providers/tester.py, lines 45-48:
```
TRANSACTION_FORMATTERS = {
    'to': apply_formatter_if(
        static_return(None),
        compose(complement(bool), decode_hex),
    ),
}
```
What would be the best way to update here? Maybe:
```
TRANSACTION_FORMATTERS = {
    'to': apply_formatter_if(
        compose(complement(bool)),
        static_return(None),
        compose(decode_hex),
    ),
}
```
I also just wanted to double check that web3/providers/eth_tester/middleware.py - line 117 (`'to': apply_formatter_if(partial(operator.eq, b''), static_return(None)),`) will also be fine in that order. I'm pretty sure that I get what's going on there and that it's fine, but I figured that it doesn't hurt to check in first.

**Comment by carver** [comment]
> What would be the best way to update here? Am I able to just flip it around?

Yeah, `compose(complement(bool), decode_hex)` is a way to check for an "empty" value that matches both '' and '0x'. So that will be the test, and `static_return(None)` is the "formatter" (it ignores the input and returns `None` any time it's applied).

> I also just wanted to double check that web3/providers/eth_tester/middleware.py - line 117 ( `'to': apply_formatter_if(partial(operator.eq, b''), static_return(None)),` ) will also be fine in that order.

That's right, `partial(operator.eq, b'')` is another way to write `lambda x: x == b''`.

**Comment by carver** [comment]
Merged in #399
